### PR TITLE
doc: added basic concepts page

### PR DESCRIPTION
**Pull request by wilkinson-steve**
Added a first attempt at a "basic concepts" page, to explain how things hang together.  Needs more work, including a description on how overwrites work, but hopefully still useful as is.

### CODE REVIEW

File: docs/docs/basic-concepts.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            "src/MyProject.Abc/*.csproj",
+            "src/MyProject.Xyz/*.csproj"
+          ],
+          "src": "path/to/csprojs"
+        }
+      ],
+      "dest": "api"
+    }
+  ],
+```
```
[the remark this answers is not in the document]
Comment (wilkinson-steve) [reply docs/docs/basic-concepts.md:58]: This is not intended to be a complete JSON object; it is intended to show a fragment of `docfx.json`.  Perhaps I should include a '`...`' to make that clearer?

It obviously doesn't make sense to include the full `docfx.json` file in the text - I think the article is verbose enough already?

File: docs/docs/basic-concepts.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+docfx build path/to/docfx.json
+```
+
+(You can append `--serve` to this step and Docfx will start a local web server so you can preview the final output.)
+
+Internally, there are many parts to this step, but in short, Docfx does the following during the `build` step:
+
+* resolve all cross-references
+* convert the YAML content from the `metadata` step into a structured data format, for passing onto the template engine 
+* convert all Markdown content into HMTL
```
Comment (yufeih) [inline docs/docs/basic-concepts.md:123]:
```suggestion
* convert all Markdown content into HTML
```

File: docs/docs/basic-concepts.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (yufeih) [reply docs/docs/basic-concepts.md:58]: Consider turn `...` into a comment otherwise it'll be rendered as red error

**Review by yufeih — APPROVED** [review]
Thank you @wilkinson-steve !

**Pull request merged**